core/src/main/java/io/aiven/kafka/tieredstorage/manifest/index/AbstractChunkIndex.java

```diff
@@ -108,6 +109,17 @@ public Chunk findChunkForOriginalOffset(final int offset) {
         );
     }
 
+    @Override
+    public List<Chunk> listChunksForRange(final BytesRange bytesRange) {
+        Chunk current;
+        final var result = new ArrayList<Chunk>();
+        for (int i = bytesRange.from; i < bytesRange.to && i < originalFileSize; i += current.originalSize) {
```

Comment:
```suggestion
        for (int i = bytesRange.from; i <= bytesRange.to && i < originalFileSize; i += current.originalSize) {
```
I guess? Since `to` is inclusive.

Reply:
It's not, e.g. chukSize=10, chunk1 start position will be 11, we want to prefetch single chunk => prefetch.max.size=10, so range will be 11-21. If we change to `<=` the result will contain 2 chunks. However we can make it inclusive so the range should be 11-20 then in this example.

Reply:
Yeah, it's good to have uniformity. You probably want to use `BytesRange.ofFromPositionAndSize` instead of `BytesRange.of`.

Reply:
Fixed